src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -121,33 +128,49 @@ public void select(final int delay) {
       } else {
         commonSelector.select(delay);
       }
-      for(final SelectionKey key: commonSelector.selectedKeys()) {
-        try {
-          if(key.isAcceptable()) {
-            doServerAccept(servers.get(key.channel()));
-          } else {
-            final Client tmpClient = clients.get(key.channel());
-            if(key.isConnectable() && tmpClient != null) {
-              doClientConnect(tmpClient, commonSelector);
-              key.cancel(); //Stupid windows bug here.
-              setClientOperations(tmpClient);
-            } else if(key.isReadable()) {
-              stats.addRead(doClientRead(tmpClient, commonSelector));
-              final Server server = servers.get(key.channel());
-              if(server != null && server.getServerType() == WireProtocol.UDP) {
-                server.acceptChannel((DatagramChannel)server.getSelectableChannel());
+      if(isRunning()) {
+        for(final SelectionKey key: commonSelector.selectedKeys()) {
+          try {
+            if(key.isAcceptable()) {
+              doServerAccept(servers.get(key.channel()));
+            } else {
+              final Client tmpClient = clients.get(key.channel());
+              try{
+                if(key.isConnectable() && tmpClient != null) {
+                  doClientConnect(tmpClient, commonSelector);
+                  key.cancel(); //Stupid windows bug here.
+                  setClientOperations(tmpClient);
+                } else {
+                  if (key.isReadable()) {
+                    if(tmpClient != null){
+                      stats.addRead(doClientRead(tmpClient, commonSelector));
+                    } else {
+                      final Server server = servers.get(key.channel());
+                      if(server != null && server.getServerType() == WireProtocol.UDP) {
+                        server.acceptChannel((DatagramChannel)server.getSelectableChannel());
+                      }
+                    }
+                  } 
+                  if(key.isWritable()) {
+                    stats.addWrite(doClientWrite(tmpClient, commonSelector));
+                  }
+                }
+              } catch(Exception e) {
```

Comment:
yeah looking at this I think I need to revert it.  I did something similar in ThreadedSE because key.isConnectable() could throw a canncelledKeyException which we are handling here already.  I will fix these up.